javascript/javascript1/week3/notes.js

```diff
@@ -0,0 +1,58 @@
+var notes = [];
+
+function saveNote(content, id){
+    if(typeof(id) === "number" && typeof(content) === "string" && content !=""){
+        notes.push({content, id});
+    } else{
+        return "Incorrect note";
+    }
+}
+
+function getNote(id){
+    if(typeof(id) === "number"){
+        for (let i = 0; i < notes.length; i++) {
+            if(notes[i].id === id){
+                return notes[i];
+            }
+            return "Error! Incorrect number."
+        }
+    } else{
+        return "Error!"
+    }
+}
+
+function logOutNotesFormatted() {
+    for (let i = 0; i < notes.length; i++) {
+        console.log(`The note with id: ${notes[i].id}, has the following note text: ${notes[i].content}`);
+    }
+}
+
+
+
+saveNote("Pick up groceries", 1);
+saveNote("Do laundry", 2);
+
+console.log(notes); // [{content: 'Pick up groceries', id: 1}, {content: 'Do laundry', id: 2}]
+
+const firstNote = getNote(1);
+console.log(firstNote); // {content: 'Pick up groceries', id: 1}
+
+
+  
+logOutNotesFormatted(); // should log out the text below
+  
+// The note with id: 1, has the following note text: Pick up groceries
+// The note with id: 2, has the following note text: Do laundry
+
+function findByWord(word){
+    const regex = new RegExp(`\\b${word}\\b`, "i");
```

Comment:
Now this is an interesting feature! 😀 Exciting to dip into Regex as well. It is a bit fragile though, e.g., if the developer puts any Regex syntax into the line it will break. (Try `findByWord(")")` and see what happens.)

To simplify things, perhaps consider looking for any notes that contain the input. Can you think of a simpler way of doing this while still making it case-insensitive? 😄 

If no notes is found, `0` is returned. This is a bit different from other methods and should be streamlined. 📏 

You are using a complex feature to find the notes matching your Regex, can you explain what `.find` does, and what `=>` means? 🔍